include/oneapi/dpl/pstl/unseq_backend_simd.h

```diff
@@ -823,7 +823,8 @@ __simd_find_first_of(_ForwardIterator1 __first, _ForwardIterator1 __last, _Forwa
     // Otherwise, vice versa.    
     if (__n1 < __n2)
     {
-        auto __u_pred = [__pred, __first](auto&& __val) { return __pred(__val, *__first); };
+        auto __u_pred =
+            [__pred, __first](auto&& __val) mutable { return __pred(std::forward<decltype(__val)>(__val), *__first); };
```

Comment:
Could you also share what was wrong before forwarding the value?

Reply:
I don't catch your question.

Reply:
> what was wrong before

A compilation error was.

Reply:
The forwarding was also present in  the replaced `__equal_value_by_pred`.